packages/api/src/logger/index.ts

```diff
@@ -0,0 +1,215 @@
+  'host',
+  'email',
+  'accessToken',
+  'access_token',
+  'secret',
+  'password',
+]
+
+/**
+ * Determines if log level based on environment variables and
+ * development or deployment environment defaults
+ *
+ * Set `LOG_LEVEL` env to the desired logging level. In order of priority, available levels are:
+ *
+ * - 'fatal'
+ * - 'error'
+ * - 'warn'
+ * - 'info'
+ * - 'debug'
+ * - 'trace'
+ *
+ * The logging level is a __minimum__ level. For instance if `logger.level` is `'info'` then all `'fatal'`, `'error'`, `'warn'`,
+ * and `'info'` logs will be enabled.
+ *
+ * You can pass `'silent'` to disable logging.
+ *
+ */
+export const logLevel: LevelWithSilent | string = (() => {
+  if (typeof process.env.LOG_LEVEL !== 'undefined') {
+    return process.env.LOG_LEVEL
+  } else if (isProduction) {
+    return 'warn'
+  } else {
+    return 'trace'
+  }
+})()
+
+/**
+ * Defines an opinionated base logger configuration that defines
+ * how to log and what to ignore.
+ *
+ * Defaults are:
+ *
+ * * Colorize output when pretty printing
+ * * Ignore certain event attributes like hostname and pid for cleaner log statements
+ * * Prefix the log output with log level
+ * * Use a shorted log message that omits server name
+ * * Humanize time in GMT
+ * * Set the default log level in dev or test to trace and warn in prod
+ *   (or set via LOG_LEVEL environment variable)
+ * * Nest objects under an `api` key to avoid conflicts with pino properties
+ * * Redact the host and other keys via a set redactionList
+ *
+ * See: https://github.com/pinojs/pino/blob/master/docs/api.md
+ *      https://github.com/pinojs/pino-pretty
+ */
+
+export const defaultLoggerOptions: LoggerOptions = {
+  prettyPrint: isPretty && {
+    colorize: true,
+    ignore: 'hostname,pid',
+    levelFirst: true,
+    messageFormat: false,
+    translateTime: true,
+  },
+  prettifier: isPretty && prettifier,
+  level: logLevel,
+  nestedKey: 'log',
+  redact: redactionsList,
+}
+
+/**
+ * RedwoodLoggerOptions
+ * Defines custom logger options that extend those available in LoggerOptions
+ * and can define a destination like a file or other supported pin log transport stream
+ *
+ * @typedef {Object} RedwoodLoggerOptions
+ * @extends LoggerOptions
+ * @property {options} LoggerOptions - options define how to log
+ * @property {string | DestinationStream} destination - destination defines where to log
+ * @property {boolean} showConfig - Display logger configuration on initialization
+ */
+export interface RedwoodLoggerOptions {
+  options: LoggerOptions
+  destination?: string | DestinationStream
+  showConfig?: boolean
+}
+
+/**
+ * Creates the logger
+ *
+ * @param {RedwoodLoggerOptions} options, destination - The logger configuration options and destination
+ * @return {BaseLogger} - The configured logger
+ *
+ * @example
+ *
+ *    createLogger({ options: { level: 'error' } })
+ */
+export const createLogger = ({
+  options,
+  destination,
+  showConfig = isDevelopment,
```

Comment:
```suggestion
  showConfig = isDevelopment,
```

Remember the todo :)